src/bcf/split.rs

```diff
@@ -92,36 +56,10 @@ enum BreakendGroup {
 }
 
 impl BreakendGroup {
-    fn subgroups<'a>(&'a self) -> Box<dyn Iterator<Item = Self> + 'a> {
-        if let BreakendGroup::Mates(mates) = self {
-            if mates.len() > 2 {
-                return Box::new(
-                    (2..mates.len())
-                        .flat_map(move |k| {
-                            if let BreakendGroup::Mates(mates) = self {
-                                mates.iter().cloned().combinations(k)
-                            } else {
-                                unreachable!();
-                            }
-                        })
-                        .map(BreakendGroup::Mates),
-                );
-            }
-        }
-        Box::new(iter::empty())
-    }
-
-    fn len(&self) -> usize {
-        match self {
-            BreakendGroup::Event(_) => 1,
-            BreakendGroup::Mates(mates) => mates.len(),
-        }
-    }
-
-    fn from(rec: &mut bcf::Record) -> Option<Self> {
-        if let Some(event) = event(rec) {
+    fn from(rec: &bcf::Record) -> Option<Self> {
+        if let Some(event) = rec.event() {
             Some(BreakendGroup::Event(event))
-        } else if let Some(mateids) = mateids(rec) {
+        } else if let Some(mateids) = rec.mateids() {
             let mut mates: Vec<_> = mateids
                 .into_iter()
                 .map(|mateid| mateid.to_owned())
```

Comment:
I think the entire collect here can now be replaced by a simple clone of the Vec.

Reply:
See clippy warning.

Reply:
Yup! (I have in-IDE on-the-fly clippy lints enabled, but this somehow didn't show up)